exec/internal/compile/backend_amd64.go

```diff
@@ -329,6 +339,43 @@ func (b *AMD64Backend) emitWasmStackLoad(builder *asm.Builder, regs *dirtyRegs,
 	prog.To.Reg = x86.REG_R13
 	builder.AddInstruction(prog)
 
+	// If bounds checked:
+	// movq r12, [r10+16]
+	// cmp r12, r13
+	// ja endbounds
+	// <emitExit() code>
+	// endbounds:
+	if b.EmitBoundsChecks {
+		prog = builder.NewProg()
+		prog.As = x86.AMOVQ
+		prog.To.Type = obj.TYPE_REG
+		prog.To.Reg = x86.REG_R12
+		prog.From.Type = obj.TYPE_MEM
+		prog.From.Reg = x86.REG_R10
+		prog.From.Offset = 16
+		builder.AddInstruction(prog)
+
+		prog = builder.NewProg()
+		prog.As = x86.ACMPQ
+		prog.To.Type = obj.TYPE_REG
+		prog.To.Reg = x86.REG_R13
+		prog.From.Type = obj.TYPE_REG
+		prog.From.Reg = x86.REG_R12
+		builder.AddInstruction(prog)
+
+		jmp := builder.NewProg()
+		jmp.As = x86.AJHI
+		jmp.To.Type = obj.TYPE_BRANCH
+		builder.AddInstruction(jmp)
+
+		b.emitExit(builder, &dirtyRegs{}, CompletionBadBounds|makeExitIndex(ci.idx))
```

Comment:
No - given the registers tracking the stack are now out of bounds, passing the value of `regs` would cause a flush of the stack to memory (corrupting it).